src/shared/Atlassian.Bitbucket.Tests/BitbucketHostProviderTest.cs

```diff
@@ -229,6 +315,81 @@ public void BitbucketHostProvider_GetSupportedAuthenticationModes(string uriStri
             Assert.Equal(expectedModes, actualModes);
         }
         
+        [Theory]
+        // DC
+        [InlineData("https", DC_SERVER_HOST, "jsquire", "password")]
+        [InlineData("http", DC_SERVER_HOST, "jsquire", "password")]
+        // cloud
+        [InlineData("https", BITBUCKET_DOT_ORG_HOST, "jsquire", "password")]
+        [InlineData("http", BITBUCKET_DOT_ORG_HOST, "jsquire", "password")]
+        public async Task BitbucketHostProvider_GetCredentialAsync_ValidateTargetUriAsync(string protocol, string host, string username, string password)
+        {
+            var input = MockInput(protocol, host, username);
+
+            var context = new TestCommandContext();
+
+            var provider = new BitbucketHostProvider(context, bitbucketAuthentication.Object, bitbucketApi.Object);
+
+            if (protocol.ToLower().Equals("http") && host.ToLower().Equals(BITBUCKET_DOT_ORG_HOST))
+            {
+                // only fail for http://bitbucket.org
+                await Assert.ThrowsAsync<Exception>(async () => await provider.GetCredentialAsync(input));
+            }
+            else
+            {
+                MockUserEntersValidBasicCredentials(bitbucketAuthentication, input, password);
+                MockRemoteBasicAuthAccountIsValidRequires2FA(bitbucketApi, input, password);
+                MockRemoteValidRefreshToken();
+                bitbucketAuthentication.Setup(m => m.ShowOAuthRequiredPromptAsync()).ReturnsAsync(true);
+                bitbucketAuthentication.Setup(m => m.CreateOAuthCredentialsAsync(It.IsAny<Uri>())).ReturnsAsync(new OAuth2TokenResult(MOCK_ACCESS_TOKEN, "access_token"));
+                var userInfo = new UserInfo() { IsTwoFactorAuthenticationEnabled = false };
+                bitbucketApi.Setup(x => x.GetUserInformationAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>())).ReturnsAsync(new RestApiResult<UserInfo>(System.Net.HttpStatusCode.OK, userInfo));
+
+                var credential = await provider.GetCredentialAsync(input);
+            }
+
+            
+        }
+
+        [Theory]
+        [InlineData("https", DC_SERVER_HOST, "jsquire")]
+        public async Task BitbucketHostProvider_StoreCredentialAsync(string protocol, string host, string username)
+        {
+            var input = MockInput(protocol, host, username);
+
+            var context = new TestCommandContext();
+
+            var provider = new BitbucketHostProvider(context, bitbucketAuthentication.Object, bitbucketApi.Object);
+
+            Assert.Equal(context.CredentialStore.Count, 0);
```

Comment:
The xunit `Assert.Equal` method's signature is `(expected, actual)`, so we should pass the `0` as the first arg (picked up by CI analysers).

```suggestion
            Assert.Equal(0, context.CredentialStore.Count);
```